### PR TITLE
[HotFix] [Seatunnel-Config] The config module easily get ConfigIncluder Class Error.

### LINKED ISSUE #4750: [Bug] [Seatunnel-Config] The config module easily get ConfigIncluder Class Error.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

When individuals repackage, it is easy to trigger errors caused by the shadow of the config module.

### SeaTunnel Version

2.3.1

### SeaTunnel Config

```conf
none
```

### Running Command

```shell
none
```

### Error Exception

```log
com.typesafe.config.ConfigIncluder无法转 换org.apache.seatunnel.shade.co.typesafe.config.ConfigInclude
```

### Flink or Spark Version

none

### Java or Scala Version

none

### Screenshots

c9fb644f3ab2f89a3165b727cce1d6f

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mosence**
Closes #4750

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Can you provide your package command? I naver meet this error.

**Comment by mosence** [comment]
> Can you provide your package command? I naver meet this error.

The first Download release-src dist, not develop environment, it mean not jar 'seatunel-config' in local maven repository,  unzip this, when call command `mvn package`  not `mvn install `, the config module dependency jar not found or some shade's Class cannot cast error would happen. 

This is a common issue in the community. This Issue that has arisen in the WeChat community.

**Comment by mosence** [comment]
> Can you provide your package command? I naver meet this error.

How to fix This license check error ?

`seatunnel-config` is not internal ?

**Comment by Hisoka-X** [comment]
image
Follow this, modify `known-dependencies.txt` in tools/dependency

**Comment by mosence** [comment]
> image Follow this, modify `known-dependencies.txt` in tools/dependency

image

I found that the internal dependencies in this document need to be changed once for each version, because the version number needs to be changed every time a new version is added. And it's SNAPSHOT version.

File: tools/dependencies/known-dependencies.txt [hunk 1]
```diff
@@ -22,6 +22,8 @@ protostuff-collectionschema-1.8.0.jar
 protostuff-core-1.8.0.jar
 protostuff-runtime-1.8.0.jar
 scala-library-2.11.12.jar
+seatunnel-config-base-2.3.1.jar
+seatunnel-config-shade-2.3.1.jar
 seatunnel-jackson-2.3.2-SNAPSHOT-optional.jar
```
Comment (EricJoy2048) [inline tools/dependencies/known-dependencies.txt:27]: Can we use `seatunnel-jackson-2.3.1.jar` and `seatunnel-guava-2.3.1.jar`?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
In next version, config-shade will support serialization, so this should not be removed in main pom.

**Comment by TyrantLucifer** [comment]
I will close this pull request, if you have any problem, please send an e-mail to dev@seatunnel.apache.org and start a discussion.

**Pull request closed without merging**